fms/utils/serialization.py

```diff
@@ -475,31 +374,10 @@ def _load_partial_state_dict(
             # If TP sharding is not needed, copy the parameter
             # into the model
             if not needs_tp_sharding or tp_module is None:
-                _copy_if_present(param, tensor_value)
+                param.copy_(tensor_value, non_blocking=True)
             elif tp_module is not None:
-                # Handle TP sharding
```

Comment:
if we also have each TP module handle its own loading, we need to break the loop above (`if isinstance(tgt_mod, TPModule)` ) on finding a TPModule, as we're no longer loading one weight at a time in this case